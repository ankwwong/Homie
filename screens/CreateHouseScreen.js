import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';

class CreateHouseScreen extends Component {
  addHouse(data) {
    var key = firebase.database().ref('/Houses').push().key;
    firebase.database().ref('/Grocery').child(key).set({item:data})
  }

  render() {
    return (
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text>CreateHouse</Text>
      </View>
    )
  }

}

export default CreateHouseScreen;
