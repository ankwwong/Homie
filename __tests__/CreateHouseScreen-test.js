import 'react-native';
import React from 'react';
import CreateHouseScreen from '../screens/CreateHouseScreen';
// import
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <CreateHouseScreen />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
