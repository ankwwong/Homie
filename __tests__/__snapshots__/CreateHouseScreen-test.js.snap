// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  enabled={true}
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
        "paddingLeft": 40,
        "paddingRight": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 60,
        }
      }
    >
      CreateHouse
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Enter a Name for Your House"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "marginBottom": 20,
          "padding": 15,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#000",
          "margin": 8,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Submit
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#000",
          "margin": 8,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
         Cancel 
      </Text>
    </View>
  </View>
</View>
`;
